stacktrace.go

```diff
@@ -7,7 +7,18 @@ import (
 	"github.com/palantir/witchcraft-go-error/internal/errors"
 )
 
-func callers() *stack {
+var _ StackTrace = (*stack)(nil)
+
+type StackTrace interface {
+	fmt.Formatter
+}
+
+type StackTracer interface {
```

Comment:
Mind adding doc comments to this and `StackTrace` above, since I can see it being confusing at first what they are used for.
The `StackTracer` is more of a provider than an actual tracer, given it provides the behavior to get a stacktrace formatter.

Reply:
Done.